activerecord/lib/active_record.rb

```diff
@@ -171,6 +174,10 @@ def self.eager_load!
     ActiveRecord::AttributeMethods.eager_load!
     ActiveRecord::ConnectionAdapters.eager_load!
   end
+
+  def self.active_job?
+    defined?(ActiveJob)
```

Comment:
This can't be here. We should not check Active Job, Active Job that should register itself on Active Record.

Can we please follow https://github.com/rails/rails/pull/34901#discussion_r246473895?

Reply:
@rafaelfranca thanks for the feedback. 

Are you reffering to this portion of the comment?

> In the Active Job railtie we use a on_load(:active_record) to set the destroy job to ActiveRecord::DestroyJob.

Are you suggesting we just set a config option such as this
https://github.com/rails/rails/blob/634f167c8430366670b08f8ffbdf735d79a6930f/activejob/lib/active_job/railtie.rb#L9

Or some other approach? Would I then just switch this method to check for the config as opposed to checking for class definition?

Thanks ❤️

Reply:
Yes, that part. I think the config is in the opposite direction. Active Job should change the `active_record.destroy_later_job` and Active Record provides the job that raises if Active Job is not loaded and the job that does the right thing.

That way we don't need to check if Active Job is loaded in Active Record, neither we need to check if Active Record is loaded in Active Job.

Reply:
I was actually raising earlier, prior to trying to enqueue the job and instead when trying to load the parent class. That way if you are in the development phase you get the notice as early as possible lowering the chance that you will miss the dependency when trying to use this option.

https://github.com/rails/rails/blob/b502fea0a81868e700fec2f2856a83b6442b4501/activerecord/lib/active_record/associations/builder/association.rb#L123-L131

I could set the destroy_later_job to false in active_record and check for it not being false or do you think I should only fail when the job is enqueued and not when the developer tries to set the `dependent_destroy` option?

Reply:
I think your solution of raising early is better, so setting the config to `false` or `nil` is good enough to me.

Reply:
Great. One more question, hopefully only 1. Do I need to move my tests somewhere else or do something to fire the railtie? I haven't tried yet figured I would ask before I potentially bang my head against my desk.

Reply:
@rafaelfranca I took a stab at this. wanna sanity check I am going in the right direction?